metadata/src/test/java/org/apache/kafka/metadata/migration/CapturingMigrationClient.java

```diff
@@ -139,14 +139,18 @@ public ZkMigrationLeadershipState setMigrationRecoveryState(ZkMigrationLeadershi
 
     @Override
     public ZkMigrationLeadershipState claimControllerLeadership(ZkMigrationLeadershipState state) {
-        this.state = state;
-        return state;
+        if (state.zkControllerEpochZkVersion() == ZkMigrationLeadershipState.EMPTY.zkControllerEpochZkVersion()) {
```

Comment:
Can we just compare against -2 (or at least have a constant for this like `ZkMigrationLeadershipState.UNKNOWN_ZK_VERSION`?)